flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -349,6 +350,14 @@ class RelTimeIndicatorConverter(rexBuilder: RexBuilder) extends RelShuttle {
             isTimeIndicatorType(updatedCall.getOperands.get(0).getType) =>
           updatedCall
 
+        case BasicOperatorTable.TUMBLE_ROWTIME |
+            BasicOperatorTable.TUMBLE_PROCTIME |
+            BasicOperatorTable.HOP_ROWTIME |
+            BasicOperatorTable.HOP_PROCTIME |
+            BasicOperatorTable.SESSION_ROWTIME |
+            BasicOperatorTable.SESSION_PROCTIME if isTimeIndicatorType(updatedCall.getType) =>
+          updatedCall
```

Comment:
But you are right, it checks the return type and not the operands which is a bug.